src/lib/components/wizard.component.ts

```diff
@@ -193,28 +231,202 @@ export class WizardComponent implements OnChanges, AfterContentInit {
    */
   public ngAfterContentInit(): void {
     // add a subscriber to the wizard steps QueryList to listen to changes in the DOM
-    this.wizardSteps.changes.subscribe(changedWizardSteps => {
-      this.model.updateWizardSteps(changedWizardSteps.toArray());
+    this.wizardStepsQueryList.changes.subscribe(changedWizardSteps => {
+      this.updateWizardSteps(changedWizardSteps.toArray());
     });
 
     // initialize the model
-    this.model.disableNavigationBar = this.disableNavigationBar;
-    this.model.defaultStepIndex = this.defaultStepIndex;
-    this.model.updateWizardSteps(this.wizardSteps.toArray());
+    this.updateWizardSteps(this.wizardStepsQueryList.toArray());
     this.updateNavigationMode(this.navigationMode);
 
-    // finally reset the whole wizard state
-    this.navigation.reset();
+    // finally reset the whole wizard componennt
+    this.reset();
+  }
+
+  /**
+   * Updates the wizard steps to the new array
+   *
+   * @param wizardSteps The updated wizard steps
+   */
+  private updateWizardSteps(wizardSteps: WizardStep[]): void {
+    // the wizard is currently not in the initialization phase
+    if (this.wizardSteps.length > 0 && this.currentStepIndex > -1) {
+      this.currentStepIndex = wizardSteps.indexOf(this.wizardSteps[this.currentStepIndex]);
+    }
+
+    this._wizardSteps = wizardSteps;
   }
 
   /**
    * Updates the navigation mode for this wizard component.
    *
-   * Initially the wizard component uses the navigation mode specified in the [[navigationMode]] input
-   * or the default navigation mode if the [[navigationMode]] input is not defined.
-   * Use this method to select a different navigation mode after the wizard component is initialized.
+   * @param navigationMode The updated navigation mode
+   */
+  public updateNavigationMode(navigationMode: NavigationModeInput) {
+    this._navigation = this.navigationModeFactory.create(this, navigationMode);
+  }
+
+  /**
+   * The completeness of the wizard.
+   * If the wizard has been completed, i.e. all steps are either completed or optional, this value is true, otherwise it is false
+   */
+  public get completed(): boolean {
```

Comment:
Getter and setter methods shouldn't be located in-between other methods

Reply:
Agree, properties were a little bit scattered around the class definition. Fixed in 4df5e97.